app/src/app/components/specmath-modal/confirm-operation/confirm-operation.component.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Component, Input } from '@angular/core';
+import { DefaultsFileUploadOptions, SpecFilesUploadOptions } from 'src/shared/interfaces';
+
+@Component({
+  selector: 'app-confirm-operation',
+  templateUrl: './confirm-operation.component.html',
+  styleUrls: ['./confirm-operation.component.scss']
+})
+export class ConfirmOperationComponent {
+  defaultsFile = null;
+  specFiles = [];
+
+  @Input() defaultsFileUploadOptions?: DefaultsFileUploadOptions;
+  @Input() specFilesUploadOptions?: SpecFilesUploadOptions;
+
+  get inputFilesValid(): boolean {
```

Comment:
When you have a getter that's referenced by the template, it's potentially getting called many times.  In a case like this where you're performing these member assignments, I would typically try to do this once, ie in the constructor.  Or, put all the members behind a getter.  But repeatedly reassigning looks odd to me.

Also, at a higher level, I'm curious - why would the orchestrator (parent modal) instantiate this component if the inputs aren't valid?

Reply:
Updated it to only render when the files are valid and removed the `inputFilesValid` from component. I didn't think of doing this way when I first wrote it 💭